tests/unit/test_compose_manager.py

```
@@ -0,0 +1,420 @@
+
+        with patch.object(manager, '_create_compose_file_on_remote', new_callable=AsyncMock) as mock_create:
+            mock_create.side_effect = Exception("Write failed")
+
+            with pytest.raises(Exception, match="Write failed"):
+                await manager.write_compose_file("test-host-1", "mystack", compose_content)
+
+    async def test_write_compose_file_invalid_host(self, docker_mcp_config, mock_docker_context_manager):
+        """Test writing compose file for invalid host."""
+        manager = ComposeManager(docker_mcp_config, mock_docker_context_manager)
+
+        with pytest.raises(ValueError):
```

Ruff (0.14.4)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)